src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -853,18 +876,48 @@ def get_num_installation_patches_removed(self):
         return len(self.__installation_patches_removed)
 
     def log_truncated_patches(self):
-        """ log details of all the removed patches from status """
+        """ Log details of all the removed patches from status """
         self.composite_logger.log_debug("Count of patches removed from: [Assessment={0}] [Installation={1}]".format(self.get_num_assessment_patches_removed(), self.get_num_installation_patches_removed()))
 
+    def __set_force_truncation_on_terminal_status(self, substatus_status):
+        """ Set force truncation to overwrite NO_TRUNCATION_LOGIC_IN_SEC time frame when terminal state (success or error) and if status file has been truncated conditions are met """
+        self.__force_truncation_on = substatus_status == Constants.STATUS_SUCCESS or substatus_status == Constants.STATUS_ERROR
+
     def __get_status_payload_with_truncated_patches(self, status_file_payload_json_dumps):
         """ Get truncated status file payload when status file byte size is more than 126kb """
         status_file_size_in_bytes = self.__calc_status_size_on_disk(status_file_payload_json_dumps)  # calc complete_status_file_payload_json byte size on disk
 
         if status_file_size_in_bytes > Constants.StatusTruncationConfig.INTERNAL_FILE_SIZE_LIMIT_IN_BYTES:  # perform truncation complete_status_file byte size > 126kb
-            truncated_status_file = self.__create_truncated_status_file(status_file_size_in_bytes, status_file_payload_json_dumps)
-            status_file_payload_json_dumps = json.dumps(truncated_status_file)
+            is_truncation_allowed = self.__calc_truncation_timestamp_in_sec() > Constants.StatusTruncationConfig.NO_TRUNCATION_LOGIC_IN_SEC
+
+            if self.__force_truncation_first_time or is_truncation_allowed or self.__force_truncation_on:
+                self.composite_logger.log_verbose("[IsTruncationAllowed={0}] [IsForceTruncationOn={1}]".format(str(is_truncation_allowed), str(self.__force_truncation_on)))
+                truncated_status_file = self.__create_truncated_status_file(status_file_size_in_bytes, status_file_payload_json_dumps)
+                self.__truncated_status_file_json_dumps = json.dumps(truncated_status_file)
+
+                if is_truncation_allowed:
+                    self.__truncation_timestamp = datetime.datetime.now()  # Set timestamp to newer time
+
+                if self.__force_truncation_first_time:
+                    self.__force_truncation_first_time = False  # Set to False because first time truncation applied during timestamp < 60 sec
+
+                return self.__truncated_status_file_json_dumps
+
+            # To ensure all newly status file remain in truncated state when complete statusfile > 126kb and timestamp < 60 sec
+            if self.__truncated_status_file_json_dumps is not None and not is_truncation_allowed and not self.__force_truncation_on:
```

Comment:
This should be else if:


Code can be optimized to:

if self.__force_truncation_first_time or is_truncation_allowed or self.__force_truncation_on:
# truncate status, set self.__truncation_timestamp  and reset self.__force_truncation_first_time. Store truncated status into status_file_payload_json_dumps

else if: self.__truncated_status_file_json_dumps is not None
 status_file_payload_json_dumps = self.__truncated_status_file_json_dumps

return status_file_payload_json_dumps at the end of the function

Reply:
I immediately return the truncated_status_json_dumps in else block to avoid more code and variable assignment. 
is this okay?
<img width="716" alt="image" src="https://github.com/Azure/LinuxPatchExtension/assets/127874208/c9f4e934-fad9-4a7f-b311-7dcd261b7661">


done

Reply:
add ut for scenario, use the minimum timestamp 2 sec (delay) and reduce MIN_TRUNCATION_INTERVAL_IN_SEC to 2sec
done